core/src/main/scala/kafka/log/Log.scala

```diff
@@ -383,7 +416,7 @@ class Log(val dir: File,
         val segment = maybeRoll(validMessages.sizeInBytes)
 
         // now append to the log
-        segment.append(appendInfo.firstOffset, validMessages)
+        segment.append(appendInfo.firstOffset, appendInfo.lastOffset, maxTimestampInMessageSet, validMessages)
```

Comment:
Hmm, it seems that if assignOffsets is false, we won't set  maxTimestampInMessageSet properly? Perhaps we should initialize maxTimestampInMessageSet in analyzeAndValidateMessageSet()?